lib/babe/verify.go

```diff
@@ -277,15 +286,23 @@ func (b *verifier) verifyAuthorshipRight(header *types.Header) error {
 	// remove seal before verifying signature
 	h := types.NewDigest()
 	for _, val := range header.Digest.Types[:len(header.Digest.Types)-1] {
-		err = h.Add(val.Value())
+		digestValue, err := val.Value()
+		if err != nil {
+			return err
```

Comment:
nit (also adding index of digest type to error context would be nice)

```suggestion
			return fmt.Errorf("getting digest type value: %w", err)
```